Sources/SmokeDynamoDB/AWSDynamoDBCompositePrimaryKeyTable+updateItems.swift

```diff
@@ -222,6 +258,118 @@ public extension AWSDynamoDBCompositePrimaryKeyTable {
+            
+            return ParameterizedStatement(statement: statement)
+        }
+        
+        let transactionInput = ExecuteTransactionInput(transactStatements: entryStatements + requiredItemsStatements)
+        
+        _ = try await dynamodb.executeTransaction(input: transactionInput)
+    }
+    
+    func transactWrite<WriteEntryType: PolymorphicWriteEntry>(_ entries: [WriteEntryType]) async throws {
+        let noContraints: [EmptyPolymorphicTransactionConstraintEntry] = []
+        return try await transactWrite(entries, constraints: noContraints)
+    }
+    
+    func transactWrite<WriteEntryType: PolymorphicWriteEntry,
+                       TransactionConstraintEntryType: PolymorphicTransactionConstraintEntry>(
+                        _ entries: [WriteEntryType], constraints: [TransactionConstraintEntryType]) async throws {
+        let entryCount = entries.count + constraints.count
+            
+        if entryCount > maximumUpdatesPerTransactionStatement {
+            throw SmokeDynamoDBError.transactionSizeExceeded(attemptedSize: entryCount,
+                                                             maximumSize: maximumUpdatesPerTransactionStatement)
+        }
+        
+        do {
+            try await self.writeTransactionItems(entries, constraints: constraints)
+        } catch DynamoDBError.transactionCanceled(let exception) {
+            let reasons = try exception.cancellationReasons?.map { cancellationReason -> SmokeDynamoDBError in
+                // https://docs.aws.amazon.com/amazondynamodb/latest/APIReference/API_ExecuteTransaction.html
+                switch cancellationReason.code {
+                case "ConditionalCheckFailed":
+                    if let item = cancellationReason.item {
+                        let key: StandardCompositePrimaryKey = try DynamoDBDecoder().decode(.init(M: item))
+                        return SmokeDynamoDBError.conditionalCheckFailed(partitionKey: key.partitionKey,
+                                                                         sortKey: key.sortKey,
+                                                                         message: cancellationReason.message)
+                    } else {
+                        return SmokeDynamoDBError.transactionUnknown(code: cancellationReason.code, message: cancellationReason.message)
+                    }
+                case "ItemCollectionSizeLimitExceeded":
+                    return SmokeDynamoDBError.transactionSizeExceeded(attemptedSize: entryCount,
+                                                                      maximumSize: maximumUpdatesPerTransactionStatement)
+                case "TransactionConflict":
+                    return SmokeDynamoDBError.transactionConflict(message: cancellationReason.message)
+
+                case "ProvisionedThroughputExceeded":
+                    return SmokeDynamoDBError.transactionProvisionedThroughputExceeded(message: cancellationReason.message)
```

Comment:
Does the concrete error type affect anything? For example, `ThrottlingError` replaces this error when DDB is set to be on-demand.

Reply:
It will determine what errors are returned to the user who will need to look through the errors to determine what they want to do. I noticed that I had missed a couple of error codes and have added them. We can combine the `transactionProvisionedThroughputExceeded` and `transactionThrottling` if we think that makes sense.